### PR TITLE
Update docs to activate env before using container templates

### LINKED ISSUE #40475: `Containerize` doesn't read templates
### Steps to reproduce

I followed the [containerize documentation](https://spack.readthedocs.io/en/latest/containers.html#extending-the-jinja2-templates) exactly and it couldn't find the custom templates.

image

### Error message

```console
# spack -d containerize
==> [2023-10-12-04:51:19.340450] Imported containerize from built-in commands
==> [2023-10-12-04:51:19.340901] Imported containerize from built-in commands
==> [2023-10-12-04:51:19.405615] Reading config from file /home/hpcadmin/spack/etc/spack/defaults/config.yaml
Traceback (most recent call last):
  File "/home/hpcadmin/spack/bin/spack", line 52, in <module>
    sys.exit(main())
  File "/home/hpcadmin/spack/lib/spack/spack_installable/main.py", line 37, in main
    sys.exit(spack.main.main(argv))
  File "/home/hpcadmin/spack/lib/spack/spack/main.py", line 1022, in main
    return _main(argv)
  File "/home/hpcadmin/spack/lib/spack/spack/main.py", line 977, in _main
    return finish_parse_and_run(parser, cmd_name, env_format_error)
  File "/home/hpcadmin/spack/lib/spack/spack/main.py", line 1005, in finish_parse_and_run
    return _invoke_command(command, parser, args, unknown)
  File "/home/hpcadmin/spack/lib/spack/spack/main.py", line 646, in _invoke_command
    return_val = command(parser, args)
  File "/home/hpcadmin/spack/lib/spack/spack/cmd/containerize.py", line 48, in containerize
    recipe = spack.container.recipe(config, last_phase=args.last_stage)
  File "/home/hpcadmin/spack/lib/spack/spack/container/writers/__init__.py", line 64, in recipe
    return create(configuration, last_phase)()
  File "/home/hpcadmin/spack/lib/spack/spack/container/writers/__init__.py", line 314, in __call__
    t = env.get_template(template_name)
  File "/home/hpcadmin/spack/lib/spack/external/_vendoring/jinja2/environment.py", line 997, in get_template
    return self._load_template(name, globals)
  File "/home/hpcadmin/spack/lib/spack/external/_vendoring/jinja2/environment.py", line 958, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "/home/hpcadmin/spack/lib/spack/external/_vendoring/jinja2/loaders.py", line 125, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "/home/hpcadmin/spack/lib/spack/external/_vendoring/jinja2/loaders.py", line 214, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: CustomDockerfile
```

### Information on your system

* **Spack:** 0.21.0.dev0 (deec1b7c2e05abcd224a73c0d53470e90b7df5bf)
* **Python:** 3.6.8
* **Platform:** linux-almalinux8-cascadelake
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

### PR DESCRIPTION
**Pull request by matt-chan**
Fixes #40475

### CODE REVIEW

**Comment by matt-chan** [comment]
@alalazo please merge if this is to your liking.

File: lib/spack/docs/containers.rst [hunk 1]
```diff
@@ -485,7 +485,7 @@ after a given stage of the build. Let's consider as an example the following str
 
 containing both the custom template extension and the environment manifest file. To use a custom
 template, the environment must register the directory containing it, and declare its use under the
-``container`` configuration:
+``container`` configuration.  A user must also activate the environment or specify ``-e </opt/environment>`` when calling containerize.
```
Comment (alalazo) [inline lib/spack/docs/containers.rst]: Can we move this to line 522? I was thinking of something similar to:
```
A user can then generate the Dockerfile with the following command:

.. code-block::

   spack -e /opt/environment containerize

Note that the environment must be active for the new template directory to be considered by Spack. 
The recipe that gets generated...
```

Comment (matt-chan) [reply lib/spack/docs/containers.rst]: Yep. I like that better. I'll fix it.

**Review by alalazo — APPROVED** [review]
Thanks!

**Pull request merged**